acme/acme/client.py

```diff
@@ -569,6 +569,7 @@ def __init__(self, directory, net):
         :param .messages.Directory directory: Directory Resource
         :param .ClientNetwork net: Client network.
         """
+        net.new_nonce_url = getattr(directory, "newNonce")
```

Comment:
Modifying the `ClientNetwork` object as a side-effect of initializing `ClientV2` seems sub-optimal. Could we make this a constructor argument for `ClientNetwork`?